sonar-python-plugin/src/main/java/org/sonar/plugins/python/IPynbSensor.java

```diff
@@ -0,0 +1,79 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.api.batch.fs.FilePredicates;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.issue.NoSonarFilter;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.plugins.python.api.ProjectPythonVersion;
+import org.sonar.plugins.python.api.PythonVersionUtils;
+import org.sonar.plugins.python.indexer.PythonIndexer;
+import org.sonar.python.checks.CheckList;
+
+import static org.sonar.plugins.python.api.PythonVersionUtils.PYTHON_VERSION_KEY;
+
+public final class IPynbSensor implements Sensor {
+
+  private final PythonChecks checks;
+  private final FileLinesContextFactory fileLinesContextFactory;
+  private final NoSonarFilter noSonarFilter;
+  private final PythonIndexer indexer;
+
+  public IPynbSensor(FileLinesContextFactory fileLinesContextFactory, CheckFactory checkFactory, NoSonarFilter noSonarFilter, PythonIndexer indexer) {
+    this.checks = new PythonChecks(checkFactory)
+      .addChecks("ipython", CheckList.getChecks());
+    this.fileLinesContextFactory = fileLinesContextFactory;
+    this.noSonarFilter = noSonarFilter;
+    this.indexer = indexer;
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor
+      .onlyOnLanguage(IPynb.KEY)
+      .name("IPython Notebooks Sensor");
+  }
+
+  @Override
+  public void execute(SensorContext context) {
+    List<InputFile> pythonFiles = getInputFiles(context);
+    Optional<String> pythonVersionParameter = context.config().get(PYTHON_VERSION_KEY);
+    pythonVersionParameter.ifPresent(value -> ProjectPythonVersion.setCurrentVersions(PythonVersionUtils.fromString(value)));
+    PythonScanner scanner = new PythonScanner(context, checks, fileLinesContextFactory, noSonarFilter, indexer);
+    scanner.execute(pythonFiles, context);
+  }
+
+  private static List<InputFile> getInputFiles(SensorContext context) {
+    FilePredicates p = context.fileSystem().predicates();
+    Iterable<InputFile> it = context.fileSystem().inputFiles(p.and(p.hasLanguage(IPynb.KEY)));
+    List<InputFile> list = new ArrayList<>();
+    it.forEach(list::add);
+    return Collections.unmodifiableList(list);
```

Comment:
could be simplified:
```java
return StreamSupport.stream(context.fileSystem().inputFiles(p.and(p.hasLanguage(IPynb.KEY))).spliterator(), false)
      .collect(Collectors.toUnmodifiableList());
```

Reply:
If you're okay, I'll keep the original version, which I find a bit clearer.